apollo-router-core/src/layers/header_manipulation.rs

```diff
@@ -0,0 +1,119 @@
+use std::task::Poll;
+
+use crate::layers::header_manipulation::Operation::{
+    Insert, Propagate, PropagateAll, PropagateOrDefault, Remove,
+};
+use http::header::HeaderName;
+use http::HeaderValue;
+use tower::{Layer, Service};
+
+use crate::SubgraphRequest;
+
+#[derive(Clone)]
+pub enum Operation {
+    PropagateAll,
+    Propagate(HeaderName),
+    PropagateOrDefault(HeaderName, HeaderValue),
+    Insert(HeaderName, HeaderValue),
+    Remove(HeaderName),
```

Comment:
If modifying a header value is required, is that anticipated that a user would express that as a Remove followed by an Insert?